src/build.rs

```diff
@@ -248,6 +248,15 @@ impl<T> FieldsBuilder<T> {
     pub fn finalize(self) -> Vec<Field<MetaForm>> {
         self.fields
     }
+
+    /// Mark last field as compact, meaning that encoding/decoding should be in the [`scale_codec::Compact`] format.
+    pub fn compact(mut self) -> Self {
+        self.fields.iter_mut().last().and_then(|f| {
+            f.compact();
+            Some(f)
+        });
+        self
+    }
```

Comment:
@ascjones This is a bit terrible but when it is [used](https://github.com/paritytech/scale-info/pull/42/files#diff-2f1d807141fcd1651cee5c5469b400f87ad94fa2fef09135bb1d100355ee4fc3R225) it seems to work out fine. The alternative is to add a new parameter to `field_of()` but it's a bit opaque for readers: `Fields::named().field_of::<[u8; 32]>("thing", "[u8; 32]", true);` – wdyt?